File: pay/src/main/java/com/binance4j/pay/dto/TradeHistory.java
package com.binance4j.pay.dto;

import java.util.List;

import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;

/**
 * Pay trade history.
 * 
 * @property code    Response code.
 * @property message Response message.
 * @property data    Trades.
 * @property success Response success.
 */
@ApiModel("Pay trade history.")
data class TradeHistory(
@ApiModelProperty("Response code.")
var code:String?=null,
@ApiModelProperty("Response message.")
var message:String?=null,
@ApiModelProperty("Trades.") List<Trade> data,
@ApiModelProperty("Response success.") var success:Boolean?=null)
{
}

File: connectors/src/main/kotlin/com/binance4j/c2c/dto/TradeHistory.kt
/*
 * MIT License
 *
 * Copyright (c) 2022 Binance4j
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package com.binance4j.c2c.dto

import io.swagger.annotations.ApiModel
import io.swagger.annotations.ApiModelProperty

/**
 * C2C trades history.
 *
 * @property code Code.
 * @property message Message.
 * @property data Trades.
 * @property total Total.
 * @property success Success.
 */
@ApiModel("C2C trades history.")
data class TradeHistory(
	@ApiModelProperty("Code.")
	val code: String,
	@ApiModelProperty("Message.")
	val message: String,
	@ApiModelProperty("Trades.")
	val data: List<Trade>,
	@ApiModelProperty("Total.")
	val total: Int,
	@ApiModelProperty("Success.")
	val success: Boolean
)